mindmeld/text_preparation/text_preparation_pipeline.py

```diff
@@ -253,9 +261,13 @@ def tokenize_and_normalize(self, text):
             if not raw_token["text"]:
                 continue
             normalized_text = self._normalize_text(raw_token["text"])
-            # We tokenize the post-norm text and split the entity if possible
+            # We sub-tokenize the post-norm text and split the entity if possible
             # Ex: normalize("o'clock") -> "o clock" -> ["o", "clock"]
-            normalized_texts = [t["text"] for t in self.tokenize(normalized_text)]
+            # Skip sub-tokenization call if characters are not added/removed
+            if normalized_text.lower() == raw_token["text"].lower():
+                normalized_texts = [normalized_text]
```

Comment:
Just a quick thought. Is there any speedup merits if we maintain a lookup dictionary for (sub-)tokenization phase? Like, we can fetch the tokenization outputs of an input from a table lookup if that input sub-text is already tokenized once previously. I think this technique can help us because there can be a lot of repetitive sub-texts in our training datasets. What do you think?